### PR TITLE
Add support for swagger 2.0

### PR DESCRIPTION
**Pull request by aneeshafedo**
resolves https://github.com/ballerina-platform/openapi-tools/issues/1091

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java [hunk 1]
```diff
@@ -103,6 +103,54 @@ public void generateClient() {
...
+            } else {
+                Assert.fail("Service was not generated");
+            }
+        } catch (IOException | BallerinaOpenApiException | FormatterException e) {
+            Assert.fail("Error while generating the service. " + e.getMessage());
+        } finally {
+            deleteGeneratedFiles("openapipetstore_service.bal");
+        }
+    }
+    @Test(description = "Test Ballerina client generation for OAS 2.0")
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java:131]: Missing new line

File: openapi-cli/src/test/resources/expected_gen/petstore_client_swagger.bal [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    @deprecated
+    remote isolated function findPetsByTags(string[] tags) returns Pet[]|error {
+        string resourcePath = string `/pet/findByTags`;
+        map<anydata> queryParam = {"tags": tags};
+        map<Encoding> queryParamEncoding = {"tags": {style: FORM, explode: true}};
+        resourcePath = resourcePath + check getPathForQueryParam(queryParam, queryParamEncoding);
+        Pet[] response = check self.clientEp->get(resourcePath);
+        return response;
+    }
+    # Find pet by ID
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/expected_gen/petstore_client_swagger.bal:97]: Missing blank lines in-between remote functions in the generated client

File: openapi-cli/src/test/resources/expected_gen/petstore_service_swagger.bal [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import ballerina/http;
+
+listener http:Listener ep0 = new (443, config = {host: "petstore.swagger.io"});
+
+service /v2 on ep0 {
+    resource function put pet(@http:Payload {mediaType: ["application/json", "application/xml"]} Pet payload) returns http:BadRequest|http:NotFound|http:MethodNotAllowed {
+    }
+    resource function post pet(@http:Payload {mediaType: ["application/json", "application/xml"]} Pet payload) returns http:MethodNotAllowed {
+    }
+    resource function get pet/findByStatus(string[] status) returns Pet[]|http:BadRequest {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/expected_gen/petstore_service_swagger.bal:10]: Missing blank lines in-between resource functions in the generated service

File: openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java [hunk 4]
```diff
@@ -410,7 +410,9 @@ public static OpenAPI getOpenAPIFromOpenAPIV3Parser(Path definitionPath) throws
         ParseOptions parseOptions = new ParseOptions();
         parseOptions.setResolve(true);
         parseOptions.setFlatten(true);
-        SwaggerParseResult parseResult = new OpenAPIV3Parser().readContents(openAPIFileContent, null, parseOptions);
+
+        SwaggerParseResult parseResult = new OpenAPIParser().readContents(openAPIFileContent, null, parseOptions);
+//        SwaggerParseResult parseResultx = new OpenAPIV3Parser().readContents(openAPIFileContent, null, parseOptions);
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java:415]: Let's remove the commented line

File: openapi-cli/src/test/resources/expected_gen/petstore_client_swagger.bal (hunk 2 above)
Comment (aneeshafedo) [reply openapi-cli/src/test/resources/expected_gen/petstore_client_swagger.bal:97]: It is how the client is getting generated from the tool. There is no new line between remote functions.

File: openapi-cli/src/test/resources/expected_gen/petstore_service_swagger.bal (hunk 3 above)
Comment (aneeshafedo) [reply openapi-cli/src/test/resources/expected_gen/petstore_service_swagger.bal:10]: It is how the service is getting generated from the tool. There is no new line between resource functions.

File: openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java (hunk 1 above)
Comment (aneeshafedo) [reply openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java:132]: Added

File: openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java (hunk 4 above)
Comment (aneeshafedo) [reply openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java]: Removed

**Review by NipunaRanasinghe — COMMENTED** [review]
Shall we also add a negative test case for the swagger definition versions below 2.0?

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by lnash94 — APPROVED** [review]

**Pull request merged**